lms/services/blackboard_api/service.py

```diff
@@ -59,6 +65,24 @@ def list_files(self, course_id):
             schema=BlackboardListFilesSchema,
         ).validated_data
 
+    def public_url(self, course_id, document_url):
+        # Get just the file_id part out of the
+        # blackboard://content-resource/file_id/ URL.
+        file_id = re.search(
+            r"blackboard:\/\/content-resource\/(?P<file_id>[^\/]*)\/", document_url
+        )["file_id"]
```

Comment:
This can be done as `file_id = document_url.split("/")[3]`. I thought using a regex would be more readable as you can actually see the `blackboard://content-resource/` in the regex pattern. You can actually see what format is expected. The regex turns out to be pretty ugly when actually written out though.

It seems slightly unbalanced that it's the _schema_ that wraps the IDs from Blackboard in these URLs but it's the _service_ that unwraps them but :man_shrugging: This service + schemas are a pretty small amount of code next to each other in one directory. There are lots of ways to write them that all have pros and cons. Probably not worth too much fiddling